tests/utils.go

```diff
@@ -204,25 +204,23 @@ func StopVirtualMachineWithTimeout(vm *v1.VirtualMachine, timeout time.Duration)
 	By("Stopping the VirtualMachineInstance")
 	virtClient := kubevirt.Client()
 
-	p, err := patch.New(
-		patch.WithAdd("/spec/running", nil),
-		patch.WithAdd("/spec/runStrategy", v1.RunStrategyHalted),
-	).GeneratePayload()
-	Expect(err).NotTo(HaveOccurred())
-	updatedVM, err := virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, p, metav1.PatchOptions{})
-	Expect(err).ToNot(HaveOccurred())
+	Expect(virtClient.VirtualMachine(vm.Namespace).Stop(context.Background(), vm.Name, &v1.StopOptions{})).To(Succeed())
 
 	// Observe the VirtualMachineInstance deleted
 	Eventually(func() error {
-		_, err = virtClient.VirtualMachineInstance(updatedVM.Namespace).Get(context.Background(), updatedVM.Name, metav1.GetOptions{})
+		_, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
 		return err
-	}, timeout, 1*time.Second).Should(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"), "The vmi did not disappear")
+	}).WithTimeout(timeout).WithPolling(time.Second).Should(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"), "The vmi did not disappear")
+
 	By("VM has not the running condition")
+	var updatedVM *v1.VirtualMachine
+	var err error
 	Eventually(func() bool {
-		vm, err := virtClient.VirtualMachine(updatedVM.Namespace).Get(context.Background(), updatedVM.Name, metav1.GetOptions{})
+		updatedVM, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
 		Expect(err).ToNot(HaveOccurred())
-		return vm.Status.Ready
-	}, timeout, 1*time.Second).Should(BeFalse())
+		return updatedVM.Status.Ready
+	}).WithTimeout(timeout).WithPolling(time.Second).Should(BeFalse())
```

Comment:
This can be potentially replaced by:
`Eventually(ThisVM(vm), ...).Should(BeReady())`

However, then we will need to fetch the VM again.

Reply:
Right, so it won't be nicer. Keeping as- is.